driver-sync/src/test/functional/com/mongodb/client/unified/ValueMatcher.java

```diff
@@ -158,7 +159,12 @@ private void assertValuesMatch(final BsonValue initialExpected, @Nullable final
     private void assertExpectedType(final BsonValue actualValue, final BsonValue expectedTypes) {
         List<String> types;
         if (expectedTypes.isString()) {
-            types = singletonList(expectedTypes.asString().getValue());
+            String expectedType = expectedTypes.asString().getValue();
+            if (expectedType.equals("number")) {
```

Comment:
Discussion only. Nothing needs to be done.

I know in this specific case, `expectedfType` can't be `null`, but generally speaking, it is a good practice to put string constant to the left hand side of `equals()` method to avoid NPE.

```
if ("number".equals(expectedType)) {
```
Not a big deal for sure.